src/dag-node/addLink.js

```diff
@@ -1,39 +1,33 @@
 'use strict'
 
-const dagNodeUtil = require('./util')
-const cloneLinks = dagNodeUtil.cloneLinks
-const cloneData = dagNodeUtil.cloneData
-const toDAGLink = dagNodeUtil.toDAGLink
-const DAGLink = require('../dag-link/dagLink')
-const DAGNode = require('./dagNode')
-const create = require('./create')
+const sortLinks = require('./sortLinks')
+const DAGLink = require('../dag-link')
 
-// Intentionally keeping the `async` to signal that it may return a promise
-// eslint-disable-next-line require-await
-const asDAGLink = async (link) => {
+const asDAGLink = (link) => {
   if (DAGLink.isDAGLink(link)) {
     // It's a DAGLink instance
     // no need to do anything
     return link
   }
 
-  if (DAGNode.isDAGNode(link)) {
-    // It's a DAGNode instance
-    // convert to link
-    return toDAGLink(link, {})
+  // DAGNode.isDagNode() would be more appropriate here, but it can't be used
+  // as it would lead to circular dependencies as `addLink` is called from
+  // within the DAGNode object.
+  if (!('cid' in link ||
```

Comment:
How much does size matter in a DAGLink? The name property is optional (`|| ''`) but the size property isn't, it just uses what it's given as the second argument. So below, `link.Tsize || link.size` seems like it's assuming something useful is there? Should it be tested for?

`!('cid' in link || 'hash' in link || 'Hash' in link || 'multihash' in link) && !('Tsize' in link || 'size' in link)`

Or maybe even stricter, `!(Number.isInteger(link.Tsize) || Number.isInteger(link.size))`?

Reply:
Good point! Given that we want (and hopefully will) get rid of the `size` I dare to leave it like this for now and merge this PR as it is.